src/gui/plugins/scene3d/Scene3D.cc

```diff
@@ -2808,7 +2812,54 @@ void Scene3D::OnDropped(const QString &_drop, int _mouseX, int _mouseY)
   math::Vector3d pos = renderWindow->ScreenToScene({_mouseX, _mouseY});
 
   msgs::EntityFactory req;
-  req.set_sdf_filename(_drop.toStdString());
+  std::string dropStr = _drop.toStdString();
+  if (QUrl(_drop).isLocalFile())
+  {
+    // mesh to sdf model
+    common::rtrim(dropStr);
+
+    if (!common::MeshManager::Instance()->IsValidFilename(dropStr))
+    {
+      this->SetErrorPopupText("Only DAE, OBJ, and STL meshes are supported.");
```

Comment:
How about being a bit more verbose:

* Mentioning the path of the file that the user tried to insert (`dropStr`)
* Mentioning that Fuel URLs are also supported

Reply:
How's this: 0fd716f68afe0a4691e42b43956b40880e9de545